src/main/java/org/wso2/carbon/inbound/amazonsqs/AmazonSQSPollingConsumer.java

```diff
@@ -0,0 +1,303 @@
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.wso2.carbon.inbound.endpoint.protocol.generic.GenericPollingConsumer;
+import com.amazonaws.services.sqs.model.Message;
+import org.wso2.carbon.utils.xml.StringUtils;
+
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * AmazonSQS inbound endpoint is used to consume messages via WSO2 ESB
+ *
+ * @since 1.0.0
+ */
+public class AmazonSQSPollingConsumer extends GenericPollingConsumer {
+
+    private static final Log logger = LogFactory.getLog(AmazonSQSPollingConsumer.class.getName());
+    //The time limit to wait when polling queues for messages
+    private int waitTime;
+    //Maximum no of messages per poll
+    private int maxNoOfMessage;
+    private BasicAWSCredentials credentials;
+    private AmazonSQS sqsClient;
+    //To check the connection to the Amazon SQS Queue
+    private boolean isConnected;
+    //URL of the Amazon SQS Queue from which you want to consume messages.
+    private String destination;
+    private ReceiveMessageRequest receiveMessageRequest;
+    private String messageReceiptHandle;
+    //list of attributes need to be receive along with the message
+    private String[] attributeNames;
+    //Content type of the message
+    private String contentType;
+
+    public AmazonSQSPollingConsumer(Properties amazonsqsProperties, String name,
+                                    SynapseEnvironment synapseEnvironment, long scanInterval,
+                                    String injectingSeq, String onErrorSeq, boolean coordination,
+                                    boolean sequential) {
+        super(amazonsqsProperties, name, synapseEnvironment, scanInterval, injectingSeq,
+                onErrorSeq, coordination, sequential);
+        this.injectingSeq = injectingSeq;
+        logger.info("Starting to load the AmazonSQS Inbound Endpoint " + name);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Starting to load the AmazonSQS Properties for " + name);
+        }
+        this.destination = properties.getProperty(AmazonSQSConstant.DESTINATION);
+        //AccessKey to interact with Amazon SQS
+        String accessKey = properties.getProperty(AmazonSQSConstant.AMAZONSQS_ACCESSKEY);
+        //SecretKey to interact with Amazon SQS
+        String secretKey = properties.getProperty(AmazonSQSConstant.AMAZONSQS_SECRETKEY);
+        if (StringUtils.isEmpty(destination)) {
+            throw new SynapseException("URL for the AmazonSQS Queue is empty");
+        }
+        if (StringUtils.isEmpty(accessKey)) {
+            throw new SynapseException("Accesskey is empty");
+        }
+        if (StringUtils.isEmpty(secretKey)) {
+            throw new SynapseException("Secretkey is empty");
+        }
+        if (!StringUtils.isEmpty(properties.getProperty(AmazonSQSConstant.AMAZONSQS_SQS_WAIT_TIME))) {
```

Comment:
Use StringUtils.isNotEmpty to check if the string is not empty or not null rather than using !StringUtils.isEmpty().